### PR TITLE
Integrating all Components for sap-converged-cloud Plan Region Mapping

### PR DESCRIPTION
**Pull request by ralikio**
**Description**

Changes proposed in this pull request:

- removal of leftovers runtime_info structures,
- injection of final implementation of `DefaultConvergedCloudRegionsProvider` in broker, but only if correct configuration is supplied (main.go) no to fail old tests.
- e2e tests for schema verification,
- minor name corrections,
- parameter validation in `NewDefaultConvergedCloudRegionsProvider` method.

**Related issue(s)**

https://github.com/kyma-project/kyma-environment-broker/issues/793

### CODE REVIEW

File: cmd/broker/broker_suite_test.go [hunk 1]
```diff
@@ -127,6 +127,12 @@ func NewBrokerSuiteTest(t *testing.T, version ...string) *BrokerSuiteTest {
 	return NewBrokerSuiteTestWithConfig(t, cfg, version...)
 }
 
+func NewBrokerSuiteTestWithConvergedRegionMappings(t *testing.T, version ...string) *BrokerSuiteTest {
```
Comment (jaroslaw-pieszka) [inline cmd/broker/broker_suite_test.go:130]: ConvergedRegion is misleading and inconsistent with our naming convention. 
ConvergedCloudRegion.

File: cmd/broker/main.go [hunk 2]
```diff
@@ -482,17 +482,27 @@ func createAPI(router *mux.Router, servicesConfig broker.ServicesConfig, planVal
 	fatalOnError(err, logs)
 	logs.Infof("Number of globalAccountIds for unlimited freeemium: %d\n", len(freemiumGlobalAccountIds))
 
+	var convergedCloudRegionProvider = broker.NewOneForAllConvergedCloudRegionsProvider()
+
+	if cfg.SapConvergedCloudRegionMappingsFilePath != "" {
+		convergedCloudRegionProvider, err = broker.NewDefaultConvergedCloudRegionsProvider(cfg.SapConvergedCloudRegionMappingsFilePath, &broker.YamlRegionReader{})
+		fatalOnError(err, logs)
+		logs.Info("sap-converged-cloud plan region mappings loaded")
```
Comment (jaroslaw-pieszka) [inline cmd/broker/main.go:490]: Consider extracting value or even usage of `SapConvergedCloudPlanName`

File: cmd/broker/schema_test.go [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+	t.Run("Create catalog - test converged cloud plan not render if not region in path", func(t *testing.T) {
+		// when
+		resp := suite.CallAPI("GET", fmt.Sprintf("oauth/v2/catalog"), ``)
+
+		content, err := io.ReadAll(resp.Body)
+		assert.NoError(t, err)
+		defer resp.Body.Close()
+
+		// then
+		assert.NotContains(t, prettify(content).String(), "sap-converged-cloud")
```
Comment (jaroslaw-pieszka) [inline cmd/broker/schema_test.go:35]: Why do we use prettify in assertion? To print pretty message when it fails?

File: cmd/broker/schema_test.go [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+		err := json.Indent(&prettyJSON, content, "", "    ")
+		assert.NoError(t, err)
+		return &prettyJSON
+	}
+
+	// given
+	suite := NewBrokerSuiteTestWithConvergedRegionMappings(t)
+	defer suite.TearDown()
+
+	t.Run("Create catalog - test converged cloud plan not render if not region in path", func(t *testing.T) {
```
Comment (jaroslaw-pieszka) [inline cmd/broker/schema_test.go:26]:
```suggestion
	t.Run("Create catalog - test converged cloud plan not rendered if no region in path", func(t *testing.T) {
```

File: internal/broker/converged_region_provider.go [hunk 5]
```diff
@@ -18,7 +20,15 @@ type DefaultConvergedCloudRegionsProvider struct {
...
+		return nil, fmt.Errorf("regionConfigurationPath cannot be empty")
+	}
+
+	if reader == nil {
+		return nil, fmt.Errorf("reader cannot be nil")
+	}
+
 	regionConfiguration, err := reader.Read(regionConfigurationPath)
 	if err != nil {
 		return nil, fmt.Errorf("while unmarshalling a file with sap-converged-cloud region mappings: %w", err)
```
Comment (jaroslaw-pieszka) [inline internal/broker/converged_region_provider.go:34]: Consider `SapConvergedCloudPlanName`

File: internal/broker/converged_region_provider.go [hunk 6]
```diff
@@ -39,9 +49,24 @@ func (c *DefaultConvergedCloudRegionsProvider) GetRegions(mappedRegion string) [
...
 }
 
+func NewOneForAllConvergedCloudRegionsProvider() ConvergedCloudRegionProvider {
+	return &OneForAllConvergedCloudRegionsProvider{}
+}
+
 type OneForAllConvergedCloudRegionsProvider struct {
 }
 
 func (c *OneForAllConvergedCloudRegionsProvider) GetRegions(mappedRegion string) []string {
```
Comment (jaroslaw-pieszka) [inline internal/broker/converged_region_provider.go:59]: Should we still return single hard-wired region?

Comment (ralikio) [reply internal/broker/converged_region_provider.go:59]: Just for all KEB tests that does not regard this functionality. I focused on testing current behaviour but other tests do not "see" region mapping functionality.

File: cmd/broker/main.go [hunk 7]
```diff
@@ -482,17 +482,27 @@ func createAPI(router *mux.Router, servicesConfig broker.ServicesConfig, planVal
...
+	} else {
+		logs.Warn("sap-converged-cloud plan region mappings file path is not set, using old configuration")
```
Comment (PK85) [inline cmd/broker/main.go:492]: this defaulting is dangerous that could cause that misconfigured production could the in BTP Australia return eu-de1

Comment (ralikio) [reply cmd/broker/main.go:493]: This was introduced for backward compatibility for tests. It was easier not to modify old tests in this case. We agreed to change that later on.

**Review by jaroslaw-pieszka — APPROVED** [review]

**Pull request merged**